core-common/src/main/java/org/glassfish/jersey/internal/config/ExternalPropertiesConfigurationFactoryFeature.java

```diff
@@ -0,0 +1,153 @@
+        return provider == null ? Collections.emptyMap() : provider.getProperties();
+    }
+
+
+    /**
+     * Input Configurable object shall be provided in order to be filled with all found properties
+     *
+     * @param config Input Configurable initialised object to be filled with properties
+     * @return true if configured false otherwise
+     */
+
+    public boolean configure(Configurable config) {
+
+        if (config instanceof ExternalConfigurationModel) {
+            return false; //shall not configure itself
+        }
+
+        final Map<String, Object> properties = readExternalPropertiesMap();
+
+        properties.forEach((k, v) -> config.property(k, v));
+
+        return true;
+    }
+
+    @Override
+    public boolean configure(FeatureContext configurableContext) {
+        return configure((Configurable) configurableContext);
+    }
+
+    /**
+     * Merged config model from all found configuration models
+     *
+     * @return merged Model object with all properties
+     */
+    public ExternalConfigurationModel getConfig() {
+        final ExternalConfigurationProvider provider = mergeConfigs(getExternalConfigurations());
+        return provider == null ? null : provider.getConfiguration();
+    }
+
+    /**
+     * List of all found models as they are found by Jersey
+     *
+     * @return list of models (or empty list)
+     */
+    public List<ExternalConfigurationProvider> getExternalConfigurations() {
+        final List<ExternalConfigurationProvider> providers = new ArrayList<>();
+        final ServiceFinder<ExternalConfigurationProvider> finder =
+                ServiceFinder.find(ExternalConfigurationProvider.class);
+        if (finder.iterator().hasNext()) {
+            finder.forEach(providers::add);
+        } else {
+            providers.add(new SystemPropertiesConfigurationProvider());
+        }
+        return providers;
+    }
+
+    private ExternalConfigurationProvider mergeConfigs(List<ExternalConfigurationProvider> configurations) {
+        final Set<ExternalConfigurationProvider> orderedConfigurations = orderConfigs(configurations);
+        final Iterator<ExternalConfigurationProvider> configurationIterator = orderedConfigurations.iterator();
+        if (!configurationIterator.hasNext()) {
+            return null;
+        }
+        final ExternalConfigurationProvider firstConfig = configurationIterator.next();
+        while (configurationIterator.hasNext()) {
+            final ExternalConfigurationProvider nextConfig = configurationIterator.next();
+            firstConfig.merge(nextConfig.getConfiguration());
+        }
+
+        return firstConfig;
+    }
+
+    private Set<ExternalConfigurationProvider> orderConfigs(List<ExternalConfigurationProvider> configurations) {
+
+        final SortedSet<ExternalConfigurationProvider> sortedSet = new TreeSet<>((config1, config2) -> {
+
+            if (
```

Comment:
If there would a config without priority, adding a new config WITH a priority would not help. The solution is to check the priority and if not found set a default priority, e.g. Priorities.USER, so that a new one could win.